File: 05_binary_search.py
def binary_search(arr, key):
    low = 0
    high = len(arr) - 1

    while low <= high:
        mid = (low + high) // 2
        if arr[mid] == key:
            return mid
        elif arr[mid] < key:
            low = mid + 1
        else:
            high = mid - 1

    return -1

array = list(map(int, input('Enter numbers: ').split()))
array.sort()
key = int(input('Enter search key: '))
result = binary_search(array, key)

if result != -1:
    print(f'Found at index {result}')
else:
    print('Not found')

File: 05_binary_search.cpp
//binary search
#include<bits/stdc++.h>
using namespace std;

int binarysearch(vector<int>&arr, int target){
    int low=0,high=arr.size()-1;

    while (low<=high){
        int mid=(low+high)/2;
        if(arr[mid]==target){
            return mid;
        }
        else if(arr[mid]>target){
            high=mid+1;
        }
        else{
            low=mid+1;
        }
    }
    return -1;
}
int main(){
    int n,target;
    cout<<"Enter the number of elements: ";
    cin>>n;
    vector<int>arr(n);
    for(int &x:arr) cin>>x;

    sort(arr.begin(),arr.end());
    cout<<"Enter the target element: ";
    cin>>target;
    cout<<"The index of the target element is: "<<binarysearch(arr,target);
}